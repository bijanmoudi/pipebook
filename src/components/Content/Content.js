import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { StickyContainer, Sticky } from 'react-sticky';
import './index.scss';

const Content = ({ title, titleExtra, search, navigation, pageContent }) => {
	return (
		<StickyContainer className="content">
			<Sticky>
				{({ style, isSticky }) => (
					<div
						className={classnames('content__header', {
							'content__header--is-sticked': isSticky
						})}
						style={style}
					>
						<div className="wrapper">
							<div className="content-header__inner">
								<div className="content-header__column content-header__column--title">
									<h1>{title}</h1>
									{titleExtra}
								</div>
								{search ? (
									<div className="content-header__column content-header__column--search">
										{search}
									</div>
								) : null}
								{navigation ? (
									<div className="content-header__column content-header__column--navigation">
										{navigation}
									</div>
								) : null}
							</div>
						</div>
					</div>
				)}
			</Sticky>
			<main className="content__body">
				<div className="wrapper">
					<div className="content-body__inner">{pageContent}</div>
				</div>
			</main>
		</StickyContainer>
	);
};

Content.propTypes = {
	title: PropTypes.string.isRequired,
	titleExtra: PropTypes.oneOfType([() => null, PropTypes.element]),
	search: PropTypes.oneOfType([() => null, PropTypes.element]),
	navigation: PropTypes.oneOfType([() => null, PropTypes.element]),
	pageContent: PropTypes.element.isRequired
};

Content.defaultProps = {
	title: null,
	titleExtra: null,
	search: null,
	navigation: null,
	pageContent: null
};

export default Content;
