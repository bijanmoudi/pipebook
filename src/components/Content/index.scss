/*==============================================
=            Content Section Styles            =
==============================================*/

@import '../../assets/styles/helpers/colors';
@import '../../assets/styles/helpers/units';
@import '../../../node_modules/sass-mq/mq';

.content {
	flex-grow: 1;
	&--is-loading {
		display: flex;
		align-items: center;
		justify-content: center;
		font-size: 2rem;
	}
	&__header {
		z-index: 1;
		position: relative;
		background-color: get-color('white', 0.975);
		box-shadow: 0 rem(1) rem(3) get-color('gray-light', 0.35);
		&--is-sticked {
			.content-header__inner {
				padding: rem(5) 0;
			}
		}
	}
	&__body {
		z-index: 0;
		position: relative;
	}
	&-header {
		&__inner {
			display: flex;
			flex-wrap: wrap;
			justify-content: space-between;
			align-items: center;
			margin: 0 rem(-5);
			padding: rem(18) 0;
			transition: padding 150ms ease;
		}
		&__column {
			display: flex;
			align-items: center;
			justify-content: center;
			padding: rem(5) rem(5);
			&--search {
				& > * {
					@include mq($until: sm) {
						width: 100%;
					}
				}
				.input {
					&__element {
						width: rem(400);
						max-width: 100%;
						padding-right: em(40);
						@include mq($until: sm) {
							width: 100%;
						}
					}
				}
			}
			&:last-child {
				justify-content: flex-end;
			}
			&:first-child {
				justify-content: flex-start;
			}
			& > * {
				& + * {
					margin-left: rem(10);
				}
			}
			&--search {
				@include mq($until: sm) {
					flex-grow: 1;
					width: 100%;
				}
			}
		}
	}
	&-body {
		&__inner {
			padding-top: rem(21);
			padding-bottom: rem(68);
		}
	}
}
