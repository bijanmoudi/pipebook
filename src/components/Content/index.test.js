import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';

const props = {
	title: 'Sample Page',
	titleExtra: <a href="#">?</a>,
	search: <input name="search" type="search" placeholder="Search ..." />,
	navigation: (
		<ol>
			<li>
				<a href="#1">Page1</a>
			</li>
			<li>
				<a href="#2">Page2</a>
			</li>
		</ol>
	),
	pageContent: <p>Page content goes here ...</p>
};

describe('Testing "Content" component', () => {
	it('Rendering a simple page', () => {
		const tree = renderer
			.create(
				<Content title={props.title} pageContent={props.pageContent} />
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering a rich page', () => {
		const tree = renderer
			.create(
				<Content
					title={props.title}
					titleExtra={props.titleExtra}
					search={props.search}
					navigation={props.navigation}
					pageContent={props.pageContent}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
