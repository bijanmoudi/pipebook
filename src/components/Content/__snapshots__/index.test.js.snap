// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing "Content" component Rendering a rich page 1`] = `
<div
  className="content"
  onScroll={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <div>
    <div />
    <div
      className="content__header"
      style={Object {}}
    >
      <div
        className="wrapper"
      >
        <div
          className="content-header__inner"
        >
          <div
            className="content-header__column content-header__column--title"
          >
            <h1>
              Sample Page
            </h1>
            <a
              href="#"
            >
              ?
            </a>
          </div>
          <div
            className="content-header__column content-header__column--search"
          >
            <input
              name="search"
              placeholder="Search ..."
              type="search"
            />
          </div>
          <div
            className="content-header__column content-header__column--navigation"
          >
            <ol>
              <li>
                <a
                  href="#1"
                >
                  Page1
                </a>
              </li>
              <li>
                <a
                  href="#2"
                >
                  Page2
                </a>
              </li>
            </ol>
          </div>
        </div>
      </div>
    </div>
  </div>
  <main
    className="content__body"
  >
    <div
      className="wrapper"
    >
      <div
        className="content-body__inner"
      >
        <p>
          Page content goes here ...
        </p>
      </div>
    </div>
  </main>
</div>
`;

exports[`Testing "Content" component Rendering a simple page 1`] = `
<div
  className="content"
  onScroll={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <div>
    <div />
    <div
      className="content__header"
      style={Object {}}
    >
      <div
        className="wrapper"
      >
        <div
          className="content-header__inner"
        >
          <div
            className="content-header__column content-header__column--title"
          >
            <h1>
              Sample Page
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
  <main
    className="content__body"
  >
    <div
      className="wrapper"
    >
      <div
        className="content-body__inner"
      >
        <p>
          Page content goes here ...
        </p>
      </div>
    </div>
  </main>
</div>
`;
